dace/sdfg/analysis/cutout.py

```diff
@@ -188,17 +193,26 @@ def singlestate_cutout(cls,
         :param reduce_input_config: Whether to reduce the input configuration where possible in singlestate cutouts.
         :param symbols_map: A mapping of symbols to values to use for the cutout. Optional, only used when reducing the
                             input configuration.
+        :param preserve_guids: If True, ensures that the GUIDs of graph elements contained in the cutout remain
+                               identical to the ones in their original graph. If False, new GUIDs will be generated.
+                               False by default - if make_copy is False, this has no effect by extension.
         :return: The created SDFGCutout.
         """
         if reduce_input_config:
             nodes = _reduce_in_configuration(state, nodes, use_alibi_nodes, symbols_map)
-        create_element = copy.deepcopy if make_copy else (lambda x: x)
+
+        def clone_f(x: Union[Memlet, InterstateEdge, nd.Node, ControlFlowBlock]):
```

Comment:
why not use a lambda for both

Reply:
The goal was to be able to type-hint, which is very cumbersome with lambdas

Reply:
See above